### PR TITLE
 Canvas based solvers

### PR DESCRIPTION
**Pull request by phtaylor**
This is the first step in resolving #261
This commit adds support for Canvas graphs as solvers in Kraken. 
I have only implemented one sample operator as a Canvas graph. (DirectionConstraintSolver)
I have tested this on the FabriceHead, and can build it in both Softimage and Maya.

### CODE REVIEW

**Comment by BigRoy** [comment]
Haven't been able to play with Kraken yet (and thus not tested this), but reading this progress is stimulating! 
Love the push towards Canvas. Keep it up!

File: Python/kraken/core/objects/operators/canvas_operator.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+        # Initialize the inputs and outputs based on the given args.
+        for port in self.graphDesc['ports']:
+            portName = port['name']
+            portConnectionType = port['execPortType']
+            if 'typeSpec' in port:
+                portDataType = port['typeSpec']
+            else:
+                portDataType = '$TYPE$'
+            print "portDataType:" + portName + ":" +portDataType
```
Comment (EricTRocks) [inline Python/kraken/core/objects/operators/canvas_operator.py]: Should remove debug printing

File: Python/kraken/core/objects/operators/canvas_operator.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    # figure out how to differentiate the solver types per operator. Maybe have
+    # an attirbute array called 'klOperators' that contains sets of what we
+    # currently have setup.
+
+    def __init__(self, name, canvasPresetPath):
+        super(CanvasOperator, self).__init__(name)
+
+        self.canvasPresetPath = canvasPresetPath
+
+        print "CanvasOperator:" + canvasPresetPath
```
Comment (EricTRocks) [inline Python/kraken/core/objects/operators/canvas_operator.py]: Should remove debug printing

File: Python/kraken/core/objects/operators/canvas_operator.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        # Load the Fabric Engine client and construct the RTVal for the Solver
+        def getPresetDesc(path):
```
Comment (EricTRocks) [inline Python/kraken/core/objects/operators/canvas_operator.py]: This doesn't need to be embedded I don't think and may be useful to have as a class method

**Comment by phtaylor** [comment]
Hi @EricTRocks . I cleaned up the issues we discussed. We can merge this now I believe.

**Pull request merged**